core/src/main/java/hudson/cli/ReloadConfigurationCommand.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2015 Red Hat, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package hudson.cli;
+
+import hudson.Extension;
+import jenkins.model.Jenkins;
+import org.acegisecurity.AccessDeniedException;
+
+import java.util.logging.Logger;
+
+/**
+ * Reload everything from file system
+ *
+ * @author pjanouse
+ * @since TODO
+ */
+@Extension
+public class ReloadConfigurationCommand extends CLICommand {
+
+    private static final Logger LOGGER = Logger.getLogger(ReloadConfigurationCommand.class.getName());
+
+    @Override
+    public String getShortDescription() {
+
+        return Messages.ReloadConfigurationCommand_ShortDescription();
+    }
+
+    @Override
+    protected int run() throws Exception {
+
+        boolean errorOccurred = false;
+
+        try {
+            Jenkins.getActiveInstance().doReload();
+        } catch (AccessDeniedException e) {
+            stderr.println(e.getMessage());
+            errorOccurred = true;
+        } catch (Exception e) {
+            final String errorMsg = String.format("Unexpected exception occurred during performing clear-queue command: %s",
```

Comment:
Actually CLICommand uses this strategy already, so I can extend it only.

That work is out of scope of this PR. In addition there is a discrepancy of returned error codes so I've created JENKINS-32273, feel free to discuss there.

Note: Not full back-ward compatibility can be achieved.

I'll continue here once PR (I'm working on it currently) for JENKINS-32273 is merged.

Another step of that effort should be to refactor already implemented and merged CLI commands in the Core, should be tracked separately as well.